minibot:
  # Publish all joint states -----------------------------------
  # 100Hz is a compromise between fast reaction and CPU load
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 100


  # TrajectoryActoin Controllers ------------------------------
  # FNH: this is a single controller for all joints; of course,
  # we could also set up controllers for the individual 'chains',
  # e.g. head, left_arm, right_arm, left_leg, right_leg.
  #
  controller:
    type: effort_controllers/JointTrajectoryController
    joints:
      - HeadPan
      - HeadTilt
      - LShoulderPitch
      - LShoulderRoll
      - LElbow
      - RShoulderPitch
      - RShoulderRoll
      - RElbow
      - LHipYaw
      - LHipRoll
      - LHipPitch
      - LKnee
      - LAnklePitch
      - LAnkleRoll
      - RHipYaw
      - RHipRoll
      - RHipPitch
      - RKnee
      - RAnklePitch
      - RAnkleRoll
    allow_partial_joints_goal: true
    constraints:
        goal_time: 10
    gains:
      HeadPan:        { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      HeadTilt:       { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      LShoulderPitch: { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      LShoulderRoll:  { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      LElbow:         { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      RShoulderPitch: { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      RShoulderRoll:  { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      RElbow:         { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      LHipYaw:        { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      LHipRoll:       { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      LHipPitch:      { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      LKnee:          { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      LAnklePitch:    { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      LAnkleRoll:     { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      RHipYaw:        { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      RHipRoll:       { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      RHipPitch:      { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      RKnee:          { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      RAnklePitch:    { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
      RAnkleRoll:     { p: 200.0, i: 0.00, d: 1.0, i_clamp: 4.0 }
 
  # Individual Joint Position Controllers ---------------------------------------

  # HEAD
  HeadPanPositionController:
    type: effort_controllers/JointPositionController
    joint: HeadPan
    pid: {p: 50.0, i: 0.00, d: 0.1}

  HeadTiltPositionController:
    type: effort_controllers/JointPositionController
    joint: HeadTilt
    pid: {p: 50.0, i: 0.00, d: 0.1}

  # ARMS

  LShoulderPitchPositionController:
    type: effort_controllers/JointPositionController
    joint: LShoulderPitch
    pid: {p: 50.0, i: 0.00, d: 0.1}

  LShoulderRollPositionController:
    type: effort_controllers/JointPositionController
    joint: LShoulderRoll
    pid: {p: 50.0, i: 0.00, d: 0.1}

  LElbowPositionController:
    type: effort_controllers/JointPositionController
    joint: LElbow
    pid: {p: 50.0, i: 0.00, d: 0.1}

  RShoulderPitchPositionController:
    type: effort_controllers/JointPositionController
    joint: RShoulderPitch
    pid: {p: 50.0, i: 0.00, d: 0.1}

  RShoulderRollPositionController:
    type: effort_controllers/JointPositionController
    joint: RShoulderRoll
    pid: {p: 50.0, i: 0.00, d: 0.1}

  RElbowPositionController:
    type: effort_controllers/JointPositionController
    joint: RElbow
    pid: {p: 50.0, i: 0.00, d: 0.1}

  # Left leg

  LHipYawPositionController:
    type: effort_controllers/JointPositionController
    joint: LHipYaw
    pid: {p: 50.0, i: 0.00, d: 0.1}

  LHipRollPositionController:
    type: effort_controllers/JointPositionController
    joint: LHipRoll
    pid: {p: 50.0, i: 0.00, d: 0.1}

  LHipPitchPositionController:
    type: effort_controllers/JointPositionController
    joint: LHipPitch
    pid: {p: 50.0, i: 0.00, d: 0.1}

  LKneePositionController:
    type: effort_controllers/JointPositionController
    joint: LKnee
    pid: {p: 50.0, i: 0.00, d: 0.1}

  LAnklePitchPositionController:
    type: effort_controllers/JointPositionController
    joint: LAnklePitch
    pid: {p: 50.0, i: 0.00, d: 0.1}

  LAnkleRollPositionController:
    type: effort_controllers/JointPositionController
    joint: LAnkleRoll
    pid: {p: 50.0, i: 0.00, d: 0.1}

  # Right leg

  RHipYawPositionController:
    type: effort_controllers/JointPositionController
    joint: RHipYaw
    pid: {p: 50.0, i: 0.00, d: 0.1}

  RHipRollPositionController:
    type: effort_controllers/JointPositionController
    joint: RHipRoll
    pid: {p: 50.0, i: 0.00, d: 0.1}

  RHipPitchPositionController:
    type: effort_controllers/JointPositionController
    joint: RHipPitch
    pid: {p: 50.0, i: 0.00, d: 0.1}

  RKneePositionController:
    type: effort_controllers/JointPositionController
    joint: RKnee
    pid: {p: 50.0, i: 0.00, d: 0.1}

  RAnklePitchPositionController:
    type: effort_controllers/JointPositionController
    joint: RAnklePitch
    pid: {p: 50.0, i: 0.00, d: 0.1}

  RAnkleRollPositionController:
    type: effort_controllers/JointPositionController
    joint: RAnkleRoll
    pid: {p: 50.0, i: 0.00, d: 0.1}

# end of file
